content/docs/esc/environments/configuring-oidc.md

```diff
@@ -0,0 +1,98 @@
+
+### Pulumi ESC custom claim
+
+The default format of the subject claim for this service is:
+
+`pulumi:environments:org:<organization name>:env:<project name>/<environment name>`
+
+To enforce more granular permissions you can customize the OIDC subject claims using the `subjectAttributes` property. This is especially useful if you plan to reference subject claims within your cloud provider’s trust policy. The default prefix when using `subjectAttributes` will be:
+
+`pulumi:environments:pulumi.organization.login:{ORGANIZATION_NAME}`
+
+Additional options for customization include:
+
+- `rootEnvironment.name`: the name of the environment that is opened first. This root environment in turn opens other imported environments
+- `currentEnvironment.name`: the name of the environment where the ESC login provider and `subjectAttributes` are defined
+- `pulumi.user.login`: the login identifier of the user opening the environment
+- `pulumi.organization.login`: the login identifier of the organization
+
+## Example: Resolving rootEnvironment.name and currentEnvironment.name
+
+Consider the following definitions for two environments, `Project/Environment-A` and `Project/Environment-B`:
+
+```yaml
+#Project/Environment-A
+values:
+  enva-rootEnv: ${context.rootEnvironment.name}
+  enva-currentEnv: ${context.currentEnvironment.name}
+
+#Project/Environment-B
+imports:
+- Project/EnvironmentA
+values:
+  envb-rootEnv: ${context.rootEnvironment.name}
+  envb-currentEnv: ${context.currentEnvironment.name}
+```
+
+When you open `Project/Environment-B`, the output would be:
+
+```
+{
+  "enva-currentEnv-name": "Project/Environment-A",
+  "enva-rootEnv-name": "Project/Environment-B",
+  "envb-currentEnv": "Project/Environment-B",
+  "envb-rootEnv": "Project/Environment-B"
+}
+```
+
+Notice how `enva-rootEnv-name` is resolved to `Project/Environment-B`. That's because Project/Environment-A is opened from Project/Environment-B which is the root, i.e. the top-level environment to be opened.
+
+When importing multiple environments into Pulumi IaC Stack Config, each environment is resolved separately. For example, if you import multiple environments into your Pulumi Stack with `rootEnvironment.name` attribute defined in all of them, then each `rootEnvironment.name` will resolve to the environment name where it is defined.
+
+## Configuring trust relationships
+
+As part of the process that exchanges your service's OIDC token for cloud provider credentials, the cloud provider must check the OIDC token's claims against the conditions configured in the provider's trust relationship. The configuration of a trust relationship varies depending on the cloud provider, but typically uses at least the **Audience**, **Subject**, and **Issuer** claims. These claims can be used to restrict trust to specific organizations, projects, stacks, environments:
```

Comment:
Where we refer to the various claims, link those to the OIDC reference documentation here: https://openid.net/specs/openid-connect-core-1_0.html#Claims.